src/manifolds/Sphere.jl

```diff
@@ -461,3 +471,103 @@ function vector_transport_to!(::AbstractSphere, Y, p, X, q, ::ParallelTransport)
     Y .= X .- m .* factor
     return Y
 end
+
+"""
+    StereographicAtlas()
+
+The stereographic atlas of ``S^n`` with two charts: one with the singular
+point (-1, 0, ..., 0) (called `:north`) and one with the singular
+point (1, 0, ..., 0) (called `:south`).
+"""
+struct StereographicAtlas <: AbstractAtlas end
+
+function get_chart_index(::Sphere{n,ℝ}, ::StereographicAtlas, p) where {n}
+    if p[1] < 0
+        return :south
+    else
+        return :north
+    end
+end
+
+function get_point_coordinates!(
+    ::Sphere{n,ℝ},
+    x,
+    ::StereographicAtlas,
+    i::Symbol,
+    p,
+) where {n}
+    if i === :north
+        return x .= p[2:end] ./ (1 + p[1])
+    else
+        return x .= p[2:end] ./ (1 - p[1])
+    end
+end
+
+function get_point!(::Sphere{n,ℝ}, p, ::StereographicAtlas, i::Symbol, x) where {n}
+    xnorm2 = dot(x, x)
+    if i === :north
+        p[1] = (1 - xnorm2) / (xnorm2 + 1)
+    else
+        p[1] = (xnorm2 - 1) / (xnorm2 + 1)
+    end
+    p[2:end] .= 2 * x / (xnorm2 + 1)
+    return p
+end
+
+function get_coordinates!(
+    ::Sphere{n,ℝ},
+    Y,
+    p,
+    X,
+    B::InducedBasis{ℝ,TangentSpaceType,<:StereographicAtlas},
+) where {n}
+    if B.i === :north
+        for i in 1:n
+            Y[i] = X[i + 1] / (1 + p[1]) - X[1] * p[i + 1] / (1 + p[1])^2
+        end
+    else
+        for i in 1:n
+            Y[i] = X[i + 1] / (-1 + p[1]) - X[1] * p[i + 1] / (-1 + p[1])^2
+        end
+    end
+    return Y
+end
+
+function get_vector!(
+    M::Sphere{n,ℝ},
+    Y,
+    p,
+    X,
+    B::InducedBasis{ℝ,TangentSpaceType,<:StereographicAtlas},
+) where {n}
+    x = get_point_coordinates(M, B.A, B.i, p)
+    mult = inv(1 + dot(x, x))^2
+
+    Y[1] = 0
+    for j in 1:n
+        Y[1] -= 4 * x[j] * mult * X[j]
+    end
+    for i in 2:(n + 1)
+        Y[i] = 0
+        for j in 1:n
+            if i == j + 1
+                Y[i] += 2 * (1 + dot(x, x) - 2 * x[i - 1]^2) * mult * X[j]
+            else
+                Y[i] -= 4 * x[i - 1] * x[j] * mult * X[j]
+            end
+        end
+        if B.i === :south
+            Y[i] *= -1
+        end
+    end
+    return Y
+end
+
+function local_metric(
+    M::Sphere{n,ℝ},
+    B::InducedBasis{ℝ,TangentSpaceType,StereographicAtlas,Symbol},
+    p,
+) where {n}
+    x = get_point_coordinates(M, B.A, B.i, p)
+    return (4 / (1 + dot(x, x))^2) * I
```

Comment:
Does `I` here “just work”? Can Julia infer its size automatically?

Reply:
"Just working" may depend on the particular operation. If Julia can infer the size from other arguments of an operation, it usually does so:
```julia
julia> a = 2*I
UniformScaling{Int64}
2*I

julia> a * [1 2; 3 4]
2×2 Array{Int64,2}:
 2  4
 6  8
```

Reply:
Ah ok, and since we think that this would only be used in these cases it should be fine – ok.